lab_3/main.py

```diff
@@ -0,0 +1,300 @@
+
+        :return:
+        """
+
+        # Filter by gender
+        if self.args.sort.lower() == 'male':
+            self.sorted_data = sorted(self.data, key=lambda x: x['gender'], reverse=True)
+        elif self.args.sort.lower() == 'female':
+            self.sorted_data = sorted(self.data, key=lambda x: x['gender'], reverse=False)
+
+        # Filter by number of rows
+        elif self.args.sort.isdigit():
+            self.sorted_data = self.data[:int(self.args.sort)]
+
+        logging.info(f'Data successfully sorted by '
+                     f'{self.args.sort.lower()} {"rows" if self.args.sort.isdigit() else "gender"}.')
+        logging.debug(f'Sorted data: {self.sorted_data}')
+
+    def add_and_change_some_data(self):
+        """
+        This method adds new columns to the data, fills the 'global.index' column with incremental indices,
+        adds the 'current.time' column with the current time adjusted by the user's timezone, and performs
+        some transformations on certain columns such as 'name.title', 'dob.date', and 'registered.date'.
+
+        :return:
+        """
+
+        # Add global index
+        self.fieldnames.append('global.index')
+        self.fieldnames.append('current.time')
+        logging.info('New columns added to fieldnames.')
+
+        for i, row in enumerate(self.sorted_data, start=1):
+            row['global.index'] = i
+        logging.info('Column "global.index" filled successfully.')
+
+        for i in self.sorted_data:
+            # Add current time
+            timezone_str = i['location.timezone.offset']
+            hours, minutes = map(int, timezone_str.split(':'))
+            i['current.time'] = timedelta(hours=hours, minutes=minutes) + datetime.now(pytz.utc)
+            logging.debug(f'Current time of user {i["login.username"]} is {i["current.time"]}.')
+
+            # Change name.title
+            if i['name.title'] == 'Mr':
+                i['name.title'] = 'Mister'
+            elif i['name.title'] == 'Ms':
+                i['name.title'] = 'Miss'
+            elif i['name.title'] == 'Mrs':
+                i['name.title'] = 'Missis'
+            elif i['name.title'] == 'Madame':
+                i['name.title'] = 'Mademoiselle'
+            logging.debug(f'Name title for user {i["login.username"]} has been changed to {i["name.title"]}.')
+
+            # Convert dob.date
+            dob_date = datetime.fromisoformat(i['dob.date'].replace('Z', '+00:00'))
+            i['dob.date'] = dob_date.strftime('%m/%d/%Y')
+            logging.debug(f'Dob date for user {i["login.username"]} has been converted to {i["dob.date"]}.')
+
+            # Convert registered.date
+            register_datetime = datetime.fromisoformat(i['registered.date'].replace('Z', '+00:00'))
+            i['registered.date'] = register_datetime.strftime('%m-%d-%Y, %H:%M:%S')
+            logging.debug(f'Register date for user {i["login.username"]} has been converted to {i["registered.date"]}.')
```

Comment:
in this **for** you perform 4 separate actions. I would advise you to move them out to separate functions